### PR TITLE
HBASE-24081 Provide documentation for running Yetus with HBase

### PR DESCRIPTION
**Pull request by ndimiduk**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m  5s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1392 |
| Optional Tests |  |
| uname | Linux 137066e23f24 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d318ca1741 |
| Max. process+thread count | 61 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 45s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 36s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   3m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1392 |
| Optional Tests |  |
| uname | Linux 4a9e03de6c25 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d318ca1741 |
| Max. process+thread count | 47 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 47s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  Maven dependency ordering for patch  |
| -0 :warning: |  whitespace  |   0m  0s |  The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   3m 34s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1392 |
| Optional Tests | dupname asflicense markdownlint |
| uname | Linux 2f72c43f681a 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d318ca1741 |
| whitespace | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/1/artifact/yetus-general-check/output/whitespace-eol.txt |
| Max. process+thread count | 47 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: dev-support/HOW_TO_YETUS_LOCAL.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# How To Use Apache Yetus with Apache HBase
+
+This document describes how to get up and running with [Apache Yetus][yetus],
+as pertains to development of Apache HBase. Specifically this covers the use
```
Comment (saintstack) [inline dev-support/HOW_TO_YETUS_LOCAL.md]: s/to/to the/

File: dev-support/HOW_TO_YETUS_LOCAL.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+of `test-patch`, of which HBase developers make use for tasks related to code
+quality. These are normally run via automation in the foundation's
+[Build][builds] infrastructure. They can also be run locally, which is the
```
Comment (saintstack) [inline dev-support/HOW_TO_YETUS_LOCAL.md]: link?

File: dev-support/HOW_TO_YETUS_LOCAL.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+subject this document.
```
Comment (saintstack) [inline dev-support/HOW_TO_YETUS_LOCAL.md]: s/this/of this/

File: dev-support/HOW_TO_YETUS_LOCAL.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+The Yetus project provides it's own documentation of `test-patch` in
```
Comment (saintstack) [inline dev-support/HOW_TO_YETUS_LOCAL.md]: s/it's/its/

File: dev-support/HOW_TO_YETUS_LOCAL.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+[Basic PreCommit][yetus-basic-precommit]. By comparison, this document is
+intended to be highly abbreviated, hands-on, and focused on the HBase use-case.
+See that document for more complete explanations and further details.
+
+## Installation
+
+In order to run Yetus, you'll need to first install Yetus and its dependencies.
+This is somewhat simplified when used in Docker mode. Yetus can be retrieved
+from a [distribution artifact][yetus-downloads]. Homebrew/Linuxbrew users can
```
Comment (saintstack) [inline dev-support/HOW_TO_YETUS_LOCAL.md]: Do these links work?

File: dev-support/HOW_TO_YETUS_LOCAL.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+"test patch." The patch can come from a number of sources, including a patch
+file, a JIRA ID, a Pull Request, &c. Or, explicitly inform Yetus that no patch
+file is provided, and the repository should be checked as is, by passing
+`--empty-patch`.
+
+## Personalities
+
+`test-patch` is extremely extensible. Even the functionality of its core
+modules can be extended or overridden. It allows for this type of
+"personalization" by way of "personality" override. Yetus ships a number of
```
Comment (saintstack) [inline dev-support/HOW_TO_YETUS_LOCAL.md]: Change s/way of/way of a/ or pluralize override.

File: dev-support/HOW_TO_YETUS_LOCAL.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+for MacOS users, there are a number of known issues with running Docker on OSX,
+so it may be preferable to instead run outside of the container.
+
+To run in Docker, of course you must install `docker` or some container runtime
+equivalent. [Docker Desktop][docker-desktop] is an option for Mac and Windows
+users.
+
+When running `test-patch` with HBase in docker mode, two parameters must be
+provided. `--docker` activates the `docker` module, enabling the feature.
+Additionally, `--dockerfile` points Yetus as HBase's provided `Dockerfile`,
```
Comment (saintstack) [inline dev-support/HOW_TO_YETUS_LOCAL.md]: s/as/at/

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   1m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1392 |
| Optional Tests |  |
| uname | Linux 7e15f73031c8 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d318ca1741 |
| Max. process+thread count | 65 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/2/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 40s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 41s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   3m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/2/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1392 |
| Optional Tests |  |
| uname | Linux 29e98d6fa99c 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d318ca1741 |
| Max. process+thread count | 46 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/2/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: dev-support/HOW_TO_YETUS_LOCAL.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+To run just the SpotBugs check, we must explicitly specify `spotbugs` and its
+dependency using `--plugins=maven,spotbugs`.
+
+## Without Docker
+
+Putting it all together, without using docker:
+
+```shell script
+$ cd /path/to/hbase
+$ JAVA_HOME=/path/to/jdk-8 test-patch.sh \
```
Comment (saintstack) [inline dev-support/HOW_TO_YETUS_LOCAL.md]: test-patch.sh is in the PATH? That an expectation?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 40s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 43s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| -0 :warning: |  whitespace  |   0m  0s |  The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   3m 36s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1392 |
| Optional Tests | dupname asflicense markdownlint |
| uname | Linux 76beda80dea5 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / d318ca1741 |
| whitespace | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/2/artifact/yetus-general-check/output/whitespace-eol.txt |
| Max. process+thread count | 47 (vs. ulimit of 10000) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1392/2/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: dev-support/HOW_TO_YETUS_LOCAL.md (hunk 2 above)
Comment (ndimiduk) [reply dev-support/HOW_TO_YETUS_LOCAL.md]: The links are pulled out to the bottom of the doc.

File: dev-support/HOW_TO_YETUS_LOCAL.md (hunk 8 above)
Comment (ndimiduk) [reply dev-support/HOW_TO_YETUS_LOCAL.md]: Yeah, the installation section says "go install this". Pulling it in from the homebrew tap places `test-patch` on the `PATH`.

File: dev-support/HOW_TO_YETUS_LOCAL.md (hunk 5 above)
Comment (ndimiduk) [reply dev-support/HOW_TO_YETUS_LOCAL.md:19]: Yes they appear to. I'm not sure if this is GH-specific syntax, but click on the "Rich Diff" icon (up by the "viewed" check box) and you'll see the rendered version of this file.

**Pull request merged**